File: 3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.java
import java.util.*;

class Solution {
    public int minMirrorPairDistance(int[] nums) {
        Map<Integer, Integer> lookup = new HashMap<>();
        int result = Integer.MAX_VALUE;

        for (int i = 0; i < nums.length; i++) {
            if (lookup.containsKey(nums[i])) {
                result = Math.min(result, i - lookup.get(nums[i]));
            }
            lookup.put(reverse(nums[i]), i);
        }

        return result != Integer.MAX_VALUE ? result : -1;
    }

    private int reverse(int n) {
        int result = 0;
        while (n > 0) {
            result = result * 10 + n % 10;
            n /= 10;
        }
        return result;
    }
}

File: leetcode/3761-minimum-absolute-distance-between-mirror-pairs/3761-minimum-absolute-distance-between-mirror-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minMirrorPairDistance = function(nums) {
    const lookup = new Map();
    let result = Number.MAX_SAFE_INTEGER;

    const reverse = (n) => {
        let r = 0;
        while (n > 0) {
            r = r * 10 + (n % 10);
            n = Math.floor(n / 10);
        }
        return r;
    };

    for (let i = 0; i < nums.length; i++) {
        if (lookup.has(nums[i])) {
            result = Math.min(result, i - lookup.get(nums[i]));
        }
        lookup.set(reverse(nums[i]), i);
    }

    return result !== Number.MAX_SAFE_INTEGER ? result : -1;
};